src/main/java/org/jboss/totp/TOTPMIDlet.java

```diff
@@ -133,8 +131,9 @@
 			TextField.NUMERIC);
 	private final TextField tfDigits = new TextField("Number of digits", String.valueOf(DEFAULT_DIGITS), 2,
 			TextField.NUMERIC);
-	private final TextField tfDelta = new TextField("Time correction (sec)", String.valueOf(DEFAULT_DELTA), 6,
-			TextField.NUMERIC);
+	//http://docs.oracle.com/javame/config/cldc/ref-impl/midp2.0/jsr118/javax/microedition/lcdui/TextField.htm getMaxSize
+	private final TextField tfDelta = new TextField("Time correction (sec)", String.valueOf(DEFAULT_DELTA), 20,
+			TextField.ANY);
```

Comment:
`TextField.NUMERIC` internally uses int type for store and checks. So user can type only 10 characters (2^32-1) of number.